### PR TITLE
ci: fix release_to_aur.yml

### PR DESCRIPTION
**Pull request by androw**
## What does this PR do
This PR change the action used to update AUR because the original one is not available on Github anymore.

## Standards checklist

- [X] The PR title is descriptive.
- [X] I have read `CONTRIBUTING.md`
- [ ] *Optional:* I have tested the code myself

I haven't been able to test myself as it's linked to topgrade CI.

### CODE REVIEW

File: .github/workflows/release_to_aur.yml
Comment (SteveLauC) [inline .github/workflows/release_to_aur.yml:19]: Looks like `v` is the default value for this parameter:

> The prefix of the version used in the tag. (v by default).

I guess this is the reason why it was not needed before 🤔

**Comment by SteveLauC** [comment]
Thanks for the fix, really appreciate it!

For the windows CI failure, you can rebase your branch once #880 is merged:)

File: .github/workflows/release_to_aur.yml
Comment (androw) [reply .github/workflows/release_to_aur.yml:19]: I think the fork added this parameter. v is marked as default but the parameter is marked as required.

**Review by SteveLauC — APPROVED** [review]
Thx!

**Comment by androw** [comment]
https://github.com/topgrade-rs/topgrade/actions/runs/11202404690/job/31138475242

Looks like there is still a problem. Is the SSH key up to date?

**Comment by SteveLauC** [comment]
> Looks like there is still a problem. Is the SSH key up to date?


Yeah, @s34m is the maintainer of the Topgrade package on AUR, I hope we can have him and give the SSH key a check: https://github.com/topgrade-rs/topgrade/issues/844#issuecomment-2395452327

**Comment by s34m** [comment]
I will check this out

**Comment by SteveLauC** [comment]
> I will check this out

Thanks, appreciate it! 😄

**Comment by s34m** [comment]
I have regenerated the ssh key in the AUR and for the Github Action

**Comment by SteveLauC** [comment]
> I have regenerated the ssh key in the AUR and for the Github Action

Thanks!

**Comment by SteveLauC** [comment]
Hi @s34m, could you please take another look at this? I just released v16.0.1 and we are still having the issue: https://github.com/topgrade-rs/topgrade/actions/runs/11285446857

**Comment by androw** [comment]
@s34m you do not seem to be listed as a maintainer: https://aur.archlinux.org/packages/topgrade

**Comment by SteveLauC** [comment]
> https://aur.archlinux.org/packages/topgrade

Actually, we have 3 Topgrade packages on AUR, as listed [here](https://repology.org/project/topgrade/versions). @s34m seems to be maintainer of [this one](https://aur.archlinux.org/packages/topgrade-bin), which is the most frequently updated one.

**Comment by androw** [comment]
It would be ideal if both topgrade and topgrade-bin are updated by GH actions

**Comment by SteveLauC** [comment]
Mind telling me why? Do they have any difference? If not, I think one would be sufficient. 👀

**Comment by androw** [comment]
Ah yes of course sorry.
topgrade: the file provides instructions to build from source and install
topgrade-bin: the file provides instructions to download the binary from GitHub and install it

Some users rebuild package, and some prefer to use the binary from upstream.

**Comment by SteveLauC** [comment]
> Some users rebuild package, and some prefer to use the binary from upstream.

Makes sense to me, we should probably reach out to the maintainer of topgrade-bin for this.

**Comment by s34m** [comment]
It seems like I am no longer maintainer of the topgrade package, probably due to the CD pipeline failing and the package being orphaned because of that.
Which is a shitty situation.

**Comment by SteveLauC** [comment]
But the package has been updated pretty frequently, who did the update then 👀

**Comment by androw** [comment]
@yochananmarqos might help us on that as he is listed as maintainer

**Comment by SinTan1729** [comment]
I think there might be a confusion. I've been the maintainer of the `topgrade-bin` package since 2022. I created the package because I just wanted a way to install `topgrade` without building it myself. As far as I know, the package didn't exist on AUR before I created it. Anyway, I'm willing to add comaintainters who are part of the project.

**Comment by SteveLauC** [comment]
Hi @SinTan1729, thanks for being here! Also, thanks @androw for leaving that comment on AUR.

>  Anyway, I'm willing to add comaintainters who are part of the project.

That would be great, could you please add @s34m as a con-maintainer so that we can release `topgrade-bin` to AUR in our CI.

**Comment by SteveLauC** [comment]
I guess it is also "s34m", as indicated by this comment, though I can not click the user name and check the profile(because I am not logged in?), so not 100% sure, let wait for @s34m to answer the question. 🫣

Screenshot 2024-10-12 at 3 19 16 PM

**Comment by yochananmarqos** [comment]
@s34m your access has been restored.

**Comment by s34m** [comment]
@SinTan1729 @yochananmarqos Thanks you two, and just to confirm it, my username is s34m.

**Comment by SinTan1729** [comment]
You've been added as a comaintainer.

**Comment by SteveLauC** [comment]
Thanks for all the help here!

So our [release_to_aur](https://github.com/topgrade-rs/topgrade/blob/main/.github/workflows/release_to_aur.yml) CI should work now, which will release the package [topgrade](https://aur.archlinux.org/packages/topgrade), and @s34m's token can also be used to release [topgrade-bin](https://aur.archlinux.org/packages/topgrade-bin), do we need to set up another CI for this?

**Comment by SteveLauC** [comment]
> do we need to set up another CI for this?

Done in #944

**Comment by SinTan1729** [comment]
@SteveLauC AUR publication seems have failed for 16.0.3
https://github.com/topgrade-rs/topgrade/actions/runs/14240423521

**Comment by androw** [comment]
Looks like the aur workflow was run before the create_release_assets_cross one.
They also do not have the same run condition.
I think we should either align run condition and order them, or we split the bin and src aur release.

**Comment by SteveLauC** [comment]
The topgrade-bin package publication failed:

```
curl: (22) The requested URL returned error: 404
  ==> ERROR: Failure while downloading https://github.com/topgrade-rs/topgrade/releases/download/v16.0.3/topgrade-v16.0.3-armv7-unknown-linux-gnueabihf.tar.gz
      Aborting...
  ==> ERROR: Failed to generate new checksums

```

> Looks like the aur workflow was run before the create_release_assets_cross one.


Yeah, that should be the root cause of that 404 request


> They also do not have the same run condition.

Mind elaborating?

**Comment by androw** [comment]
In release_to_aur.yml, the run condition is:
```
  push:
    tags:
      - "v*"
```

In create_release_assets_cross.yml, the run condition is:
```
  release:
    types: [created]
```

AUR is run as soon as the tag is pushed while building occurs when the release is created.

Maybe we can change the run condition in AUR to something like:
```
  workflow_run:
    workflows: ["Publish release files for non-cd-native environments"]
    types:
      - completed
```

**Comment by SteveLauC** [comment]
> ```
>  workflow_run:
>    workflows: ["Publish release files for non-cd-native environments"]
>    types:
>      - completed
> ```

Yeah, this makes sense, let's do this!

**Comment by SteveLauC** [comment]
From the CI log(see the binaries it tried to download):

```
==> Retrieving sources...
    -> Downloading topgrade_x86_64-16.0.3.tar.gz...
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  
    0 5315k    0 20652    0     0  85497      0  0:01:03 --:--:--  0:01:03 85497
  100 5315k  100 5315k    0     0  19.6M      0 --:--:-- --:--:-- --:--:--  235M
    -> Downloading topgrade_aarch64-16.0.3.tar.gz...
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  
  100 49[29](https://github.com/topgrade-rs/topgrade/actions/runs/14240423521/job/39908861278#step:4:30)k  100 4929k    0     0  13.5M      0 --:--:-- --:--:-- --:--:-- 13.5M
    -> Downloading topgrade_armv7h-16.0.3.tar.gz...
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (22) The requested URL returned error: [40](https://github.com/topgrade-rs/topgrade/actions/runs/14240423521/job/39908861278#step:4:41)4
  ==> ERROR: Failure while downloading https://github.com/topgrade-rs/topgrade/releases/download/v16.0.3/topgrade-v16.0.3-armv7-unknown-linux-gnueabihf.tar.gz
      Aborting...
  ==> ERROR: Failed to generate new checksums
```

I guess it relies on both "Publish release files for CD native environments" and "Publish release files for non-cd-native environments" workflows🤔

**Comment by SteveLauC** [comment]
> I guess it relies on both "Publish release files for CD native environments" and "Publish release files for non-cd-native environments" workflows🤔

Shoot, looks like we [cannot wait for more than one workflows](https://github.com/orgs/community/discussions/16059)

**Pull request merged**
